custom/icds_reports/utils/aggregation_helpers/distributed/app_commcare_version.py

```diff
@@ -0,0 +1,118 @@
+from dateutil.relativedelta import relativedelta
+
+from custom.icds_reports.const import AGG_APP_VERSION_TABLE
+from custom.icds_reports.utils.aggregation_helpers import month_formatter, transform_day_to_month, \
+    get_app_version_temp_tablename
+from custom.icds_reports.utils.aggregation_helpers.distributed.base import (
+    BaseICDSAggregationDistributedHelper
+)
+
+
+class AppVersionAggregationDistributedHelper(BaseICDSAggregationDistributedHelper):
+    helper_key = 'usage-forms'
+    ucr_data_source_id = 'static-usage_forms'
+    tablename = AGG_APP_VERSION_TABLE
+
+    def __init__(self, month):
+        self.month = transform_day_to_month(month)
+        self.prev_month_start = self.month - relativedelta(months=1)
+        self.next_month_start = self.month + relativedelta(months=1)
+
+    def aggregate(self, cursor):
+        create_temp_query = self.create_temporary_table()
+        drop_temp_query = self.drop_temporary_table()
+        agg_query, agg_params = self.aggregation_query()
+        index_queries = self.indexes()
+
+        cursor.execute(drop_temp_query)
+        cursor.execute(create_temp_query)
+        cursor.execute(agg_query, agg_params)
+
+        for query in index_queries:
+            cursor.execute(query)
+
+        for i, query in enumerate(self.aggregation_queries()):
+            cursor.execute(query)
+        cursor.execute(drop_temp_query)
+
+    @property
+    def temporary_tablename(self):
+        return get_app_version_temp_tablename()
+
+    def data_from_ucr_query(self):
+        query_params = {
+            "start_date": month_formatter(self.month),
+            "end_date": month_formatter(self.next_month_start)
+        }
+        return """
+            SELECT distinct on (awc_id)
+                awc_id,
+                supervisor_id,
+                %(start_date)s::DATE AS month,
+                app_version as app_version,
+                commcare_version as commcare_version
+            FROM "{ucr_tablename}" WHERE
+                form_date >= %(start_date)s AND form_date < %(end_date)s ORDER BY 
+                awc_id, form_date  DESC
```

Comment:
Is this one row per form or awc?

Reply:
awc

Reply:
How does that happen? I do not see any grouping or other syntax that would cause aggregation by awc